src/main/java/com/google/api/codegen/config/PackageMetadataConfig.java

```diff
@@ -307,7 +320,9 @@ public static PackageMetadataConfig createFromString(String yamlContents) {
       Map<String, Object> configMap, String key) {
     Map<TargetLanguage, Map<String, VersionBound>> packageDependencies = new HashMap<>();
     List<String> packages = (List<String>) configMap.get(key);
-    Preconditions.checkArgument(packages != null, key + " is missing from package metadata config");
+    if (packages == null) {
+      return packageDependencies;
```

Comment:
Why is null an acceptable value now? Looks weird. If in discogapic flow it can be null, it is better just to provide an empty collection where appropriate (in the place which writes into `configMap`) instead. This is minor, but is still weird.

Reply:
Discogapic is missing a lot of config values (e.g. `gax_grpc_version`, `proto_deps`), and this would require many lines of `if(value == null)` checks and null handling.

The place that creates the `configMap` is in a 3rd party `org.yaml` package.

Reply:
In case if it is  gapic execution path (not discogapic) this would make `null` a valid value (used to be invalid). I understand that it is minor, but if it is possible to keep it invalid in case of gapic (not discogapic) then please make it so (but only if it is simple to do and does not complicate or "uglify" the code significantly).

Reply:
I'm not sure of a way to implement gapic-only checks and discogapic-only checks in a clean way. One solution would be to put the input source (e.g. protobuf, discovery-doc) in input yaml, and then introduce `if` statements for every case that applies to different execution paths differently. 

This PR also introduces the config attribute `gax_http_version` which will be null for gapic execution paths, so null is a valid value for any path.